sift/src/main/java/siftscience/android/AppStateCollector.java

```diff
@@ -60,15 +71,11 @@ public AppStateCollector(SiftImpl sift, Context context) {
 
         this.acquiredNewLocation = false;
         if (!sift.getConfig().disallowLocationCollection) {
-            try {
-                this.googleApiClient = new GoogleApiClient.Builder(this.context)
-                        .addApi(LocationServices.API)
-                        .addConnectionCallbacks(this)
-                        .addOnConnectionFailedListener(this)
-                        .build();
-            } catch (Exception e) {
-                Log.e(TAG, e.toString());
-            }
+            mFusedLocationClient = LocationServices.getFusedLocationProviderClient(this.context);
+            mSettingsClient = LocationServices.getSettingsClient(this.context);
+            createLocationCallback();
```

Comment:
is there any reason all of these helpers directly modify instance state versus returning built objects?

Reply:
@garylee1 Those are the initialization of helpers and there is no builder for FusedLocationProviderClient as in the GoogleApiClient. 
We followed as per the guidelines mentioned in [android/location-samples](https://github.com/android/location-samples/blob/02e5fb05c80bf2ffac430bb8a57158356de84d86/LocationUpdates/app/src/main/java/com/google/android/gms/location/sample/locationupdates/MainActivity.java#L185)